arshitcc/Taste-Trek
# fix : update the state after cookie-session expires

## What does this PR do?

- Fixes #1  (GitHub issue number)
- Fixes unauthorized application state after the session expiry

## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?

- Login with valid credentials.
<img width="1435" alt="Screenshot 2024-12-31 at 7 15 02 AM" src="https://github.com/user-attachments/assets/063dd9db-d5fc-4edc-adbc-a8e00efc9563" />

- Perform an operation in the app (e.g., add an item to the cart).
- Clear the cookies (simulate session expiry).
   
<img width="1440" alt="Screenshot 2024-12-31 at 7 15 15 AM" src="https://github.com/user-attachments/assets/323f4796-901c-4b34-a095-1c0c9ef2247c" />

- Attempt any operation like adding items to the cart or fetching data.
- User is redirected to the login page as expected and no further actions can be performed without a valid session.
<img width="1435" alt="Screenshot 2024-12-31 at 7 16 12 AM" src="https://github.com/user-attachments/assets/b4c58cd5-7811-4dc9-805d-da3c83dfe48a" />
